ert3/evaluator/_evaluator.py

```diff
@@ -31,78 +104,61 @@ def _create_evaluator_tmp_dir(workspace_root: Path, evaluation_name: str) -> Pat
-        for command in step_config.transportable_commands:
-            if storage_config.get("type") == "shared_disk":
-                transmitter = ert3.data.SharedDiskRecordTransmitter(
-                    name=command.name,
-                    storage_path=pathlib.Path(storage_config["storage_path"]),
-                )
-            else:
-                raise ValueError(
-                    f"Unsupported transmitter type: {storage_config.get('type')}"
-                )
-            with open(command.location, "rb") as f:
-                asyncio.get_event_loop().run_until_complete(
-                    transmitter.transmit_data([f.read()])
-                )
-            for iens in range(0, ensemble_size):
-                transmitters[iens][command.name] = transmitter
-    return dict(transmitters)
 
+def _function_jobs(step: Function) -> TupleDictStrAny:
+    return (
+        {
+            "name": step.function.__name__,
+            "executable": cloudpickle.dumps(step.function),
+        },
+    )
 
-def _prepare_output(
-    ee_config: Dict[str, Any],
-    step_config: Step,
-    evaluation_tmp_dir: Path,
-    ensemble_size: int,
-) -> Dict[int, Dict[str, RecordTransmitter]]:
-    tmp_input_folder = evaluation_tmp_dir / "output_files"
-    os.makedirs(tmp_input_folder)
-    storage_config = ee_config["storage"]
-    transmitters: Dict[int, Dict[str, ert3.data.RecordTransmitter]] = defaultdict(dict)
-
-    for output in step_config.output:
-        for iens in range(0, ensemble_size):
-            if storage_config.get("type") == "shared_disk":
-                transmitters[iens][
-                    output.record
-                ] = ert3.data.SharedDiskRecordTransmitter(
-                    name=output.record,
-                    storage_path=pathlib.Path(storage_config["storage_path"]),
-                )
-            else:
-                raise ValueError(
-                    f"Unsupported transmitter type: {storage_config.get('type')}"
-                )
-    return dict(transmitters)
+
+def _unix_jobs(step: Unix) -> TupleDictStrAny:
+    def job(script: str) -> Dict[str, Any]:
+        name, *args = script.split()
+        return {
+            "name": name,
+            "executable": next(
+                (cmd.path for cmd in step.transportable_commands if cmd.name == name),
+                pathlib.Path(name),
+            ),
+            "args": tuple(args),
+        }
+
+    return tuple(map(job, step.script))
+
+
+def _name_to_record_key(
+    command: TransportableCommand,
+) -> Dict[str, Any]:
+    cmd: Dict[str, Any] = command.dict(exclude={"name"}, by_alias=True)
+    cmd["record"] = command.name
+    return cmd
+
+
+def _job_specific_fields(step: Step) -> Tuple[TupleDictStrAny, TupleDictStrAny]:
+    return (
+        (_unix_jobs(step), tuple(map(_name_to_record_key, step.transportable_commands)))
+        if isinstance(step, Unix)
+        else (_function_jobs(step), ())
+    )
```

Comment:
I think a comment would be nice here…

Reply:
And the commands are step-specific aren't they? So that function name doesn't quite make sense.

I think it's fine to make this more fleshed out/imperative in the `_step` function.

Reply:
_job_specifc_fields is changed to _ensemble_attribute_discriminator and _step is change to _ensemble_step_attributes which returns a dict instead of a tuple[dict]